pages/learn/studio/create/characters/controllers/joystick.md

```diff
@@ -0,0 +1,58 @@
+# Joystick Character Controller
+
+## Introduction
+
+**Joystick** character controllers are a type of character controller that allows the player to move a character using an on-screen joystick.
+
+<Note type="warning">
+Joystick is still in beta and may have some limitations or bugs. Please report any issues you encounter to the Highrise Studio team.
+</Note>
+
+## Adding a Joystick Character Controller
+
+Luckily, Highrise Studio comes with a built-in Joystick character controller that you can use to control your characters. Here's how you can add a Joystick character controller to your character:
+
+<Note type="warning">
+You need version "0.12.2" or later of Highrise Studio to use the Joystick character controller.
+</Note>
+
+1. **Right-click** in the project view and select `Create > Highrise > Scene Variant` to create a new scene variant.
+2. Rename the scene variant to something like "JoystickScenePrefab".
+3. In the **JoystickScenePrefab**, add a component called `JoystickCharacterController`
+4. Remove the default `PlayerCharacterController` component from the **JoystickScenePrefab**
+5. Customize the Joystick properties in the `JoystickCharacterController` component
+6. Click "Use this prefab" to set it as the default scene variant
+
+## Customizing the Joystick
+
+You can customize the Joystick character controller by adjusting the following properties in the `JoystickCharacterController` component:
+
+1. **Footstep Walk Sound**: The sound that plays when the character is walking.
+2. **Footstep Run Sound**: The sound that plays when the character is running.
+3. **External Input Action**: The input action that triggers the joystick movement.
+4. **Running Speed Threshold**: The speed at which the character switches from walking to running.
+5. **Is 2D**: Whether the joystick should move the character in 2D space.
+6. **Max Link Distance**: The maximum distance the joystick can be from the character.
+7. **Max Line Angle**: The maximum angle the joystick can be from the character.
+8. **Max Link Move Step Change**: The maximum step change when moving the joystick.
```

Comment:
These three determine when a nav mesh link can be used. All three conditions must be met to jump through a nav mesh link. Also it's "linK", not "linE" in the second one, and "mIN" not "mAX" in the third one.
1. Max Link Distance is just the in-world distance between the character and the link endpoint. If it's beyond this threshold the link won't be considered.
2. Max Link Angle is the angle between your movement vector and the link's jump direction. If the angle is larger it won't be considered.
3. Min Link Move Step Change is the difference (as a proportion) between the desired movement step (as in velocity vector * time delta), and the actual step when the nav mesh surface is taken into account. For example if you're standing on the right edge of a cliff and try to move right, the step change will be 1 as you won't be able to move at all due to being on the edge of the possible movement surface. On the other hand the value of 0 represents no change in movement -- your desired direction is completely within the nav mesh surface. This prevents jumping through nav mesh links when it's possible to just move in that direction. You will only jump once you encounter an edge (or in the platformer camera case if you point the movement vector upwards or downwards enough). A value of 0.5 here would mean the nav mesh link will only be considered if your movement is at least half restricted (e.g. you approached an edge at an angle).